tests/staggered_eigensolve_test_gtest.hpp

```diff
@@ -0,0 +1,184 @@
+  auto compute_svd = ::testing::get<3>(test_param);
+  auto spectrum = ::testing::get<4>(test_param);
+
+  // Reverse engineer the operator type
+  QudaSolveType combo_solve_type = get_solve_type(use_norm_op, use_pc, compute_svd);
+  if (combo_solve_type == QUDA_DIRECT_PC_SOLVE) {
+    // matpc
+
+    // this is only legal for the staggered and asqtad op
+    if (!is_staggered(dslash_type))
+      return true;
+
+    // we can only compute the real part for Lanczos, and real or magnitude for Arnoldi
+    switch (eig_type) {
+    case QUDA_EIG_TR_LANCZOS:
+    case QUDA_EIG_BLK_TR_LANCZOS:
+      if (spectrum != QUDA_SPECTRUM_LR_EIG && spectrum != QUDA_SPECTRUM_SR_EIG) return true;
+      break;
+    case QUDA_EIG_IR_ARNOLDI:
+      if (spectrum == QUDA_SPECTRUM_LI_EIG || spectrum == QUDA_SPECTRUM_SI_EIG) return true;
+      break;
+    default: break;
+    }
+  } else if (combo_solve_type == QUDA_NORMOP_SOLVE) {
+    // matdag_mat
+    
+    // this is only legal for the staggered and asqtad op
+    if (!is_staggered(dslash_type))
+      return true;
+
+    switch (eig_type) {
+    case QUDA_EIG_TR_LANCZOS:
+    case QUDA_EIG_BLK_TR_LANCZOS:
+      if (spectrum != QUDA_SPECTRUM_LR_EIG && spectrum != QUDA_SPECTRUM_SR_EIG) return true;
+      break;
+    case QUDA_EIG_IR_ARNOLDI:
+      //if (spectrum == QUDA_SPECTRUM_LI_EIG || spectrum == QUDA_SPECTRUM_SI_EIG) return true;
+      return true; // we skip this because it takes an unnecessarily long time and it's covered elsewhere
+      break;
+    default: return true; break;
+    }
+  } else if (combo_solve_type == QUDA_DIRECT_SOLVE) {
+    // mat
+    
+    switch (dslash_type) {
+    case QUDA_STAGGERED_DSLASH:
+      // only Arnoldi, imaginary part or magnitude works (real part is degenerate)
+      // We skip SM because it takes an unnecessarily long time and it's
+      // covered by HISQ
+      if (eig_type != QUDA_EIG_IR_ARNOLDI) return true;
+      if (spectrum != QUDA_SPECTRUM_LI_EIG && spectrum != QUDA_SPECTRUM_SI_EIG &&
+            spectrum != QUDA_SPECTRUM_LM_EIG) return true;
+      break;
+    case QUDA_ASQTAD_DSLASH:
+      // only Arnoldi, imaginary part or magnitude works (real part is degenerate)
+      if (eig_type != QUDA_EIG_IR_ARNOLDI) return true;
+      if (spectrum == QUDA_SPECTRUM_LR_EIG || spectrum == QUDA_SPECTRUM_SR_EIG) return true;
+      break;
+    case QUDA_LAPLACE_DSLASH:
+      switch (eig_type) {
+      case QUDA_EIG_TR_LANCZOS:
+      case QUDA_EIG_BLK_TR_LANCZOS:
+        if (spectrum != QUDA_SPECTRUM_LR_EIG && spectrum != QUDA_SPECTRUM_SR_EIG) return true;
+        break;
+      case QUDA_EIG_IR_ARNOLDI:
+        if (spectrum == QUDA_SPECTRUM_LI_EIG || spectrum == QUDA_SPECTRUM_SI_EIG) return true;
+        break;
+      default: return true; break;
+      }
+      break;
+    default: return true; break;
+    }
+  }
+
+  // There seems to be some stubborn issue with this combination on 2xGPUs that I can't quite
```

Comment:
Is this still a thing?

Reply:
https://github.com/lattice/quda/pull/1421/commits/1228db9e5941575175d547322d0d4e193ba49ede